### PR TITLE
Chunked bulk indexing to lower memory footprint

### LINKED ISSUE #680: Add Haystack <-> Elasticsearch-DSL converters
Add converter to transform existing requests from what Haystack expects to what Elasticsearch-DSL uses.

Add a converter to transform the results of an Elasticsearch-DSL query to what the existing front-end/API expects.

After this is merged, we can start ripping out the Haystack dependency. In the future, the code should then be cleaned up to remove this work, so we can directly use the cleaner output of Elasticsearch-DSL.

Part of #668.

### PR DESCRIPTION
**Pull request by noisecapella**
Related to #680, still need to remove Haystack to finish it

### CODE REVIEW

**Comment by noisecapella** [comment]
PR #745 must be merged before the tests will pass

File: search/utils.py [hunk 1]
```diff
@@ -47,13 +47,28 @@ def get_conn():
...
+
     # Make sure everything exists.
     if not _CONN.indices.exists(INDEX_NAME):
-        _CONN.indices.create(INDEX_NAME)
+        raise ReindexException("Unable to find index {index_name}".format(
+            index_name=INDEX_NAME
+        ))
+
+    mapping = _CONN.indices.get_mapping()
+    if INDEX_NAME not in mapping:
```
Comment (ShawnMilo) [inline search/utils.py]: This is checking for the index a second time, not the mapping. 

[This](https://github.com/mitodl/lore/pull/700#issuecomment-142340863) script contains the syntax to look `DOC_TYPE`.

File: search/tests/test_es_indexing.py [hunk 2]
```diff
@@ -255,3 +265,36 @@ def test_sorting(self):
...
+            # Mapping doesn't exist yet
+            get_conn()
+
+        conn = get_conn(verify=False)
+        conn.indices.create(INDEX_NAME)
+        conn.indices.refresh()
+        create_mapping()
+
+        with self.assertRaises(Exception):
+            self.client.get(search_url)
```
Comment (ShawnMilo) [inline search/tests/test_es_indexing.py]: Why does this raise an exception, and what is the exception type? Given that the index and mapping exist and are empty, why doesn't it just return no error and no results?

**Comment by ShawnMilo** [comment]
This looks good, and tests pass for me locally after merging master.

I refreshed the index three times, and recreated it three times locally. All six runs took 44-49 seconds and consumed 390 MB of RAM.

Good to merge.

File: search/tests/test_es_indexing.py (hunk 2 above)
Comment (noisecapella) [inline search/tests/test_es_indexing.py]: I'm not sure, I'll look into it

Comment (noisecapella) [inline search/tests/test_es_indexing.py]: I'm getting a TransportError here with `No mapping found for [id] in order to sort on`. It only goes away when I index the resources. I'll investigate a fix for this

**Comment by noisecapella** [comment]
After investigating I think the problem is that `id` is dynamically mapped and it complains when being asked to sort on a mapping which doesn't exist yet since there are no resources at first. I tried just adding `id` to the mapping but this collides with Haystack somehow. I added a workaround to set `ignore_unmapped: True` on sort by `id`. I'll file an issue to reinvestigate this workaround, I feel like we will be able to remove it after getting rid of Haystack entirely.

**Comment by ShawnMilo** [comment]
Okay. It works and tests pass, so as long an Travis completes successfully again then it's good to merge.

**Comment by noisecapella** [comment]
One test was failing due to the mock complaining about a function not being called in a management command, even though it seems like it was called and it passed locally. I changed the test to just assert the reindexing instead, will merge when they pass

**Pull request merged**